### ISSUE TITLE
Clerk branch issues, why is the clerk required?

### ISSUE THREAD

**Issue by vdboor** [machtfit/adyen#1]
I've tried applying your code to an oscar project, but got really stuck trying to port the code to the clerk branch (1.0.1-machtfit-3). There are various issues with this branch:
- handle_payment() no longer received other kwargs. Providing the clerk would help processing the payment.
- There are 2 `RedirectRequired` classes with a different signature. This breaks when loading the payment's app `RedirectRequired` into the `OrderPlacementMixin` code, which also uses checkout's app `RedirectRequired`
- There are calls to `User.handle_order_shipped_to()` thus requiring the project to switch `AUTH_USER_MODEL`.
- The enforced user no longer has a username, which breaks the my existing installation.

At the end, this is just a no-go. I really like the idea of the clerk, but adding so many invasive changes makes it impossible to use this part of the app.

**Comment by mbertheau** [comment]
Hi @vdboor, thank you for your interest in this code :)

I assume you were trying to port your oscar-based application to the clerk branch? It's true that there's a number of unfinished things in the clerk branch. Thank you for bringing up which specific problems you encountered. This will help later when and if me or someone else finds the time to adapt the clerk branch so that in can be merged into oscar.

Concerning the specific issues:
1. I don't recall the details, but the payment process is currently not as customizable as it is in stock oscar, however it is sufficient for our needs. You're welcome to 
2. I didn't notice that. This will have to be fixed.
3. Yes, I didn't think about that. That method will have to go somewhere else.
4. Resolved when 3 is resolved.

I'll note that we have an oscar application running with this version of adyen and the version of django-oscar that you mentioned, in case there's some kind of branch mixup.

It should be possible to create a python module that can be used to provide adyen payments for stock oscar, possibly using the `django_adyen` and `adyen` modules from here. A possible starting point is https://github.com/machtfit/adyen/commit/38100b74401280722d41483339b66b9a538d9575 which adapted this module to the checkout flow thingie that isn't in oscar yet. The changes in https://github.com/machtfit/adyen/commit/d736442c7c5622b42260b82089fba2fb537741d9 are also relevant for that.

**Comment by vdboor** [comment]
Thanks for the update! I'm indeed persueing what you've suggested: to write a custom python module that integrates the django_adyen code. It's still better then nothing :)
- I've noticed one other thing in the branch, the COUNT/VALUE/COVERAGE conditions were removed in the `offer.Condition` model, which breaks all standard vouchers that were already in a older database. 
- The PaymentMethodView got removed.
- The payment-details page no longer displays the error when a payment error occurs, users get a standard preview page instead.
- The `offer.Benefit` model has a new `can_apply_with_other_benefits`, for which there is no south migration.
- The Category.full_name is removed.
